dimod/bidict.py

```diff
@@ -0,0 +1,140 @@
+# Copyright 2020 D-Wave Systems Inc.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+
+from collections.abc import Mapping, MutableMapping
+
+
+__all__ = ['BiDict']
+
+
+class BiDict(MutableMapping):
+    """A BiDict is a dictionary with fast reverse lookup.
+
+    All dictionary methods are supported. In addition there is an `inverse`
+    attribute, which it itself a `BiDict`, storing the values as keys
+    and vice versa.
+
+    Attributes:
+
+        inverse:
+            Another `BiDict`, equivalent to one constructed by
+            `{value: key for key, value in d}`.
+
+    Examples:
+
+        >>> from dimod.bidict import BiDict
+
+        `BiDict` can be constructed and interacted with just like a normal
+        dictionary, but it also has an `inverse` attribute, which is itself
+        another (inversed) `BiDict`.
```

Comment:
I'd delete `BiDict` can be constructed and ... itself  another (inversed) `BiDict`.